File: driver-service/src/main/java/com/tompang/carpool/driver_service/dto/RegisterDriverRequestDto.java
package com.tompang.carpool.driver_service.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.Data;

@Data
public class RegisterDriverRequestDto {
    @NotBlank
    public String userId;
    @NotBlank
    public String vehicleRegistrationNumber;
    @NotBlank
    public String vehicleMake;
    @NotBlank
    public String vehicleModel;

    public boolean requireManualReview = false;
}

File: webapp-bff/src/backend/driver/dto/register-driver-request.dto.ts
import { IsBoolean, IsString } from "class-validator";

export class RegisterDriverRequestDto {
  @IsString()
  userId: string;

  @IsString()
  vehicleRegistrationNumber: string;
  
  @IsString()
  vehicleMake: string;
  
  @IsString()
  vehicleModel: string;
  
  @IsBoolean()
  requireManualReview: boolean = false; // default
}